public/app/features/live/centrifuge/service.ts

```diff
@@ -1,14 +1,12 @@
 import Centrifuge from 'centrifuge/dist/centrifuge';
-import { GrafanaLiveSrv, LiveDataStreamOptions } from '@grafana/runtime/src/services/live';
-import { BehaviorSubject, Observable, share, startWith } from 'rxjs';
 import {
-  LiveChannelAddress,
-  LiveChannelConnectionState,
-  LiveChannelId,
+  GrafanaLiveSrv,
+  LiveDataStreamOptions,
   StreamingFrameAction,
   StreamingFrameOptions,
-  toLiveChannelId,
-} from '@grafana/data';
+} from '@grafana/runtime/src/services/live';
```

Comment:
nit: I'm not sure we should have imports like this. I guess we should either move the types to core or expose them externally from `@grafana/runtime`.

Reply:
We ran into an issue with that - this file runs in a web worker and thus does not have access to some Browser APIs/env. Direct import, ie. `import _something_ from '@grafana/runtime'` pulls in some code which tries to e.g. access the `document` prop which results in an error

I was thinking about splitting the runtime package into two or more to solve this, maybe we should do that next? @ryantxu @mckn

Reply:
from within core, we should try to avoid imports from explicit folders -- but I don't think it matters much.  In particular this style import lets us import things that are not necessarily exported in the external pacages.

splitting runtime :grimacing: lets discuss more, hopefully we can avoid that